File: lib/location_data.dart
import 'dart:convert';

class LocationData {
  double latitude = 0;
  double longitude = 0;
  double speed = 0;
  double accuracy = 0;
  DateTime time;
  bool isFake;

  LocationData(
    this.latitude,
    this.longitude, {
    this.speed = 0,
    this.accuracy = 0,
    this.time,
    this.isFake = false,
  });

  @override
  String toString() {
    if (isFake) {
      return jsonEncode({"latitude": latitude, "longitude": longitude, "isFake": isFake});
    }
    return jsonEncode({
      "latitude": latitude,
      "longitude": longitude,
      "speed": speed.toStringAsFixed(0),
      "accuracy": accuracy.toStringAsFixed(0),
      "time": time.toString()
    });
  }
}

File: android/src/main/java/com/ahmet/real_location/LocationData.java
package com.ahmet.real_location;

import android.annotation.SuppressLint;
import android.location.Location;

import org.json.JSONObject;

import java.text.SimpleDateFormat;
import java.util.Date;

@SuppressLint("SimpleDateFormat")
public class LocationData {
    long time;
    double latitude;
    double longitude;
    double speed;
    double accuracy;
    boolean isFake;

    public LocationData(Location location, boolean isFake) {
        this.accuracy = location.getAccuracy();
        this.speed = location.getSpeed();
        this.longitude = location.getLongitude();
        this.latitude = location.getLatitude();
        this.time = location.getTime();
        this.isFake = isFake;
    }

    @Override
    public String toString() {
        JSONObject jsonObject = new JSONObject();
        try {
            jsonObject.put("latitude", latitude);
            jsonObject.put("longitude", longitude);
            jsonObject.put("speed", speed);
            jsonObject.put("accuracy", accuracy);
            jsonObject.put("time", time);
            jsonObject.put("isFake", isFake);
            jsonObject.put("time_format", longToDateString());

        } catch (Exception ignored) {
        }
        return jsonObject.toString();
    }

    private String longToDateString() {
        try {
            SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
            return format.format(new Date(time));
        } catch (Exception e) {
            return "[!]";
        }

    }
}
